### PR TITLE
Two survey fixes: emoji filtering & dashboard sorting

### PR DESCRIPTION
**Pull request by breville**
Two survey fixes:
- replace all supplementary characters before submitting, because just emoji didn't cover everything.  Details in comments in the code.
- Sort the teacher dashboard results by result text as well, so that multi results are sorted by question.

### CODE REVIEW

**Comment by breville** [comment]
With huge thanks to the multi-talented @bzotto for reasoning through the character encoding issue.

File: apps/src/code-studio/levels/levelGroup.js [hunk 1]
```diff
@@ -145,20 +145,27 @@ window.initLevelGroup = function (
...
-      String.fromCharCode(0xd83d) + '[' + String.fromCharCode(0xde80) + '-' + String.fromCharCode(0xdeff) + ']'
-    ];
+    // Build the range for the supplementary pair in a way that works with Babel
+    // (which currently handles \u encoding in a string incorrectly).
+    var range =
+      '[' + String.fromCharCode(0xD800) + '-' + String.fromCharCode(0xDBFF) + '][' +
+            String.fromCharCode(0xDC00) + '-' + String.fromCharCode(0xDFFF) + ']';
 
-    return source.replace(new RegExp(ranges.join('|'), 'g'), blankCharacter);
+    return source.replace(new RegExp(range, 'g'), blankCharacter);
```
Comment (mehalshah) [inline apps/src/code-studio/levels/levelGroup.js:168]: nit - I feel like you can initialize range inline here.

**Comment by mehalshah** [comment]
one nit, otherwise LGTM

File: apps/src/code-studio/levels/levelGroup.js (hunk 1 above)
Comment (breville) [inline apps/src/code-studio/levels/levelGroup.js:168]: fair point, but I think that might become a bit hard to read, just from a formatting point of view.  Would be much easier if we didn't have to do this workaround for the Babel bug.

**Pull request merged**